src/Product/WCProductAdapter.php

```diff
@@ -903,4 +907,149 @@ public function setTargetCountry( $targetCountry ) {
 		// product shipping information is also country based
 		$this->map_wc_product_shipping();
 	}
+
+	/**
+	 * Performs the attribute mapping.
+	 * This function applies rules setting values for the different attributes in the product.
+	 *
+	 * @param array $mapping_rules The set of rules to apply
+	 */
+	protected function map_attribute_mapping_rules( array $mapping_rules ) {
+		$attributes = [];
+
+		if ( empty( $mapping_rules ) ) {
+			return $this;
+		}
+
+		foreach ( $mapping_rules as $mapping_rule ) {
+			if ( $this->rule_match_conditions( $mapping_rule ) ) {
+				$attribute_id                = $mapping_rule['attribute'];
+				$attributes[ $attribute_id ] = self::format_attribute( apply_filters( "woocommerce_gla_product_attribute_value_{$attribute_id}", $this->get_source( $mapping_rule['source'] ), $this->get_wc_product() ), $attribute_id );
```

Comment:
We can call `$this->format_attribute` it's not a static function.

Formatting it across multiple lines makes it easier to see what it's doing:
```suggestion
				$attributes[ $attribute_id ] = self::format_attribute(
					apply_filters(
						"woocommerce_gla_product_attribute_value_{$attribute_id}",
						$this->get_source( $mapping_rule['source'] ),
						$this->get_wc_product()
					),
					$attribute_id
				);
```

Also I had a question about the filter name, this allows us to override the mapped value, but as far as I understand if it's also set in the product data, then the filter will be called twice. Should we filter the `mapping_rules` with `gla_attributes` (which is a list of all attributes which are set in the product)? Technically we can skip mapping rules if we know they are going to be overwritten right after.